python/lsst/meas/algorithms/gp_interpolation.py

```diff
@@ -0,0 +1,657 @@
+    factor = (jax.scipy.linalg.cholesky(kernel, overwrite_a=True, lower=False), False)
+    alpha = jax.scipy.linalg.cho_solve(factor, y, overwrite_b=False)
+    return alpha.reshape((len(alpha), 1))
+
+
+@jit
+def jax_get_y_predict(kernel_rect, alpha):
+    """
+    Compute the predicted values of y using the given kernel and alpha.
+
+    Parameters:
+    -----------
+    kernel_rect : array-like
+        The kernel matrix.
+    alpha : array-like
+        The alpha vector.
+
+    Returns:
+    --------
+    array-like
+        The predicted values of y.
+
+    """
+    return jnp.dot(kernel_rect, alpha).T[0]
+
+
+class GaussianProcessJax:
+    def __init__(self, std=1.0, correlation_length=1.0, white_noise=0.0, mean=0.0):
+        self.std = std
+        self.correlation_lenght = correlation_length
+        self.white_noise = white_noise
+        self.mean = mean
+        self._alpha = None
+
+    def fit(self, x_good, y_good):
+        y = y_good - self.mean
+        self._x = x_good
+        kernel = jax_rbf_kernel(x_good, self.std, self.correlation_lenght, self.white_noise)
+        self._alpha = jax_get_alpha(y, kernel)
+
+    def predict(self, x_bad):
+        kernel_rect = jax_rbf_kernel_rect(x_bad, self._x, self.std, self.correlation_lenght)
+        y_pred = jax_get_y_predict(kernel_rect, self._alpha)
+        return y_pred + self.mean
+
+
+# Vanilla Gaussian Process regression using treegp package
+# There is no fancy O(N*log(N)) solver here, just the basic GP regression (Cholesky).
+class GaussianProcessTreegp:
+    """
+    Gaussian Process Treegp class for Gaussian Process interpolation.
+
+    Parameters:
+    -----------
+    std : float, optional
+        Standard deviation of the Gaussian Process kernel. Default is 1.0.
+    correlation_length : float, optional
+        Correlation length of the Gaussian Process kernel. Default is 1.0.
+    white_noise : float, optional
+        White noise level of the Gaussian Process. Default is 0.0.
+    mean : float, optional
+        Mean value of the Gaussian Process. Default is 0.0.
+
+    Methods:
+    --------
+    fit(x_good, y_good):
+        Fit the Gaussian Process to the given training data.
+
+        Parameters:
+        -----------
+        x_good : array-like
+            Input features for the training data.
+        y_good : array-like
+            Target values for the training data.
+
+    predict(x_bad):
+        Predict the target values for the given input features.
+
+        Parameters:
+        -----------
+        x_bad : array-like
+            Input features for the prediction.
+
+        Returns:
+        --------
+        y_pred : array-like
+            Predicted target values.
+
+    """
+
+    def __init__(self, std=1.0, correlation_length=1.0, white_noise=0.0, mean=0.0):
+        self.std = std
+        self.correlation_length = correlation_length
+        self.white_noise = white_noise
+        self.mean = mean
+
+    def fit(self, x_good, y_good):
```

Comment:
As I commented elsewhere, I would use more neutral terms, like `x_train` and `x_predict` or similar, instead of `x_good` and `x_bad`.